File: src/main/java/com/gitlab/pedrioko/core/zk/component/ProgressTask.java
package com.gitlab.pedrioko.core.zk.component;

import com.gitlab.pedrioko.core.view.api.OnEvent;
import com.gitlab.pedrioko.core.zk.component.model.enums.ProgressTaskType;
import org.zkoss.zk.ui.HtmlBasedComponent;
import org.zkoss.zk.ui.event.Events;
import org.zkoss.zk.ui.sys.ContentRenderer;

import java.io.IOException;

public class ProgressTask extends HtmlBasedComponent {

    private long time = 1000;
    private OnEvent onTimerEvent = null;
    private long maxValue = 0;
    private long currentValue = 0;
    private ProgressTaskType type;

    public ProgressTask() {
        super();
        addEventListener(Events.ON_AFTER_SIZE, e -> {
            if (onTimerEvent != null)
                onTimerEvent.doSomething();
        });
    }

    @Override
    protected void renderProperties(ContentRenderer renderer) throws IOException {
        super.renderProperties(renderer);
        render(renderer, "time", getTime());
        render(renderer, "maxValue", getMaxValue());
        render(renderer, "currentValue", getCurrentValue());
        render(renderer, "type", getType());
    }

    public long getMaxValue() {
        return maxValue;
    }

    public void setMaxValue(long maxValue) {
        this.maxValue = maxValue;
        smartUpdate("maxValue", getMaxValue());

    }

    public long getCurrentValue() {
        return currentValue;
    }



    public void setCurrentValue(long currentValue) {
        if (this.currentValue != currentValue) {
            this.currentValue = currentValue;
            smartUpdate("currentValue", getCurrentValue());
        }
    }

    public ProgressTaskType getType() {
        return type;
    }

    public void setType(ProgressTaskType type) {
        this.type = type;
        smartUpdate("type", getType());

    }

    public long getTime() {
        return time;
    }

    public void setTime(long time) {
        this.time = time;
        smartUpdate("time", getTime());
    }

    public OnEvent getOnTimerEvent() {
        return onTimerEvent;
    }

    public void setOnTimerEvent(OnEvent onTimerEvent) {
        this.onTimerEvent = onTimerEvent;
    }
}

File: ats-gitlab-core/src/main/resources/web/js/zk/progresstask/ProgressTask.js
zk.progresstask.ProgressTask = zk.$extends(zk.Widget, {

    _time: '',
    _maxValue: '',
    _type: '',
    _currentValue: '',
    _bar: '',
    _timeout: '',
    getMaxValue: function () {
        return this._maxValue;
    },
    setMaxValue: function (src) {
        if (this._maxValue != src) {
            this._maxValue = src;
        }
    },
    getCurrentValue: function () {
        return this._currentValue;
    },
    setCurrentValue: function (src) {
        if (this._currentValue != src) {
            this._currentValue = src;
        }
    },
    getType: function () {
        return this._type;
    },
    setType: function (src) {
        if (this._type != src) {
            this._type = src;
        }
    },
    getTime: function () {
        return this._time;
    },
    setTime: function (src) {
        if (this._time != src) {
            this._time = src;
        }
    },
    bind_: function () {
        this.$supers('bind_', arguments);


        var _bar = new ldBar("#"+this.uuid + '-progresstask');
        /*console.log(_bar);
        function doSomething() {
            $(this.uuid + '-progresstask').fire('onAfterSize',{}, {toServer:true})
            _bar.set(this._currentValue);
        }*/

        if (_bar) {
            //this._timeout = setInterval(doSomething, this.getTime());
            _bar.set(this._currentValue);
            // this._bar.style = this.getStyle()
           this._bar.style.width = this.getWidth();
           this._bar.style.height = this.getHeight();
        }
    },
    unbind_: function () {
        this.$supers('unbind_', arguments);
    },
    domContent_: function () {
        return "";
    },
});